### DISCUSSION TITLE
How much memory does vector required?

### DISCUSSION THREAD

**Discussion by jmjoy** [vectordotdev/vector#17779] [General]
I found that our vector process often has OOM, using kafka as the source and elasticsearch as the sink.

I'm not sure if it's caused by a memory leak or a large amount of data.

**Comment by LinTechSo** [comment]
Hi, any updates ?
Same in vector version 0.29.1
source[kafka] --> sink[elastic]

my configuration:
```
memory buffer
adaptive request concurrency
max_events: 500
memory request resource: 1Gi
```

i know based on events rate and size, memory usage should increased but i want to know how can i prevent vector  from OOMKill in the case of high rate
